src/contracts/edition.ts

```diff
@@ -91,25 +85,14 @@ export class Edition extends Erc1155<TokenERC1155> {
    * ```
    */
   public royalties: ContractRoyalty<TokenERC1155, typeof Edition.schema>;
-  /**
-   * Signature Minting
-   * @remarks Generate dynamic NFTs with your own signature, and let others mint them using that signature.
-   * @example
-   * ```javascript
-   * // see how to craft a payload to sign in the `contract.signature.generate()` documentation
-   * const signedPayload = contract.signature.generate(payload);
-   *
-   * // now anyone can mint the NFT
-   * const tx = contract.signature.mint(signedPayload);
-   * const receipt = tx.receipt; // the mint transaction receipt
-   * const mintedId = tx.id; // the id of the NFT minted
-   * ```
-   */
-  public signature: Erc1155SignatureMinting;
+
   /**
    * @internal
    */
   public interceptor: ContractInterceptor<TokenERC1155>;
+  private _query = this.query as Erc1155Enumerable;
+  private _mint = this.mint as Erc1155Mintable;
+  private _batchMint = this._mint.batch as Erc1155BatchMintable;
```

Comment:
need to re-expose signature as always defined here, i think this can work:

```
public signature = this.signature as Erc1155SignatureMintable;
```

Reply:
This worked well with the instance in the base class